quix-frontend/client/src/state-components/history/HistoryComponent.tsx

```diff
@@ -107,69 +98,27 @@ export function History(props: HistoryProps) {
     }
   }
 
-  const renderInitialState = () => (
-    <div className="bi-c-h bi-align bi-center bi-grow">
-      <div className="bi-empty-state--loading bi-fade-in">
-        <div className="bi-empty-state-content" data-hook="history-initial">Loading query history...</div>
-      </div>
-    </div>
-  );
-
-  const renderFilterInitialState = () => (
-    <div className="bi-c-h bi-align bi-center bi-grow">
-      <div className="bi-empty-state--loading bi-fade-in">
-        <div className="bi-empty-state-content" data-hook="history-initial">Searching query history...</div>
-      </div>
-    </div>
-  );
-
-  const renderErrorState = () => (
-    <div className="bi-c-h bi-align bi-center bi-grow">
-      <div className="bi-empty-state bi-fade-in">
-        <div className="bi-empty-state-icon bi-danger">
-          <i className="bi-icon bi-danger">error_outline</i>
-        </div>
-        <div className="bi-empty-state-header" data-hook="history-error">{stateData.errorMessage}</div>
-      </div>
-    </div>
-  );
-
-  const renderEmptyState = () => (
-    <div className="bi-c-h bi-align bi-center bi-grow">
-      <div className="bi-empty-state bi-fade-in">
-        <img className="bi-empty-state-image" src={noResultsImg}></img>
-        <div className="bi-empty-state-header" data-hook="history-result">No results</div>
-      </div>
-    </div>
-  );
-
   const renderContentState = () => (
-    <div
-      className="bi-panel bi-c-h bi-fade-in bi-theme--lighter"
-      data-hook="history-content"
-    >
-      <div className="bi-panel-content bi-c-h">
-        <Table
-          initialData={stateData.rows}
-          loadMore={getChunk}
-          onRowClicked={onHistoryClicked}
-          columns={historyTableFields.map(field => ({
-            Header: field.title,
-            Cell: table => field.filter(undefined, table.row.original, 0, highlightQuery(field.name)),
-            accessor: field.name,
-            className: field.className,
-          }))}
-          paginationSize={CHUNK_SIZE}
-          tableSize={(size) => viewState.update({ size })}
-        />
-      </div>
-    </div>
+    <PaginatedTable
+      entityName="history"
```

Comment:
this component shouldn't be aware of what entity it is, it should be completely generic.

Reply:
it is a generic component. makePaginantion has required props and any other props, the reason that there is 'entityName' is for data-hooks

Reply:
in this case you can change the prop name to "hookName"